jsonschema-core.md

```diff
@@ -1271,10 +1266,10 @@ the `$id` of the embedded or referenced resource unless it is specifically
 desired to identify the object containing the `$ref` in the second
 (non-embedded) arrangement.
 
-An implementation MAY choose not to support addressing schema resource contents
-by IRIs using a base other than the resource's canonical IRI, plus a JSON
-Pointer fragment relative to that base. Therefore, schema authors SHOULD NOT
-rely on such IRIs, as using them may reduce interoperability.[^8]
+Due to the potential break in functionality described above, the behavior for
+using JSON Pointer fragments that point to or cross a resource boundary is
+undefined.  Schema authors SHOULD NOT rely on such IRIs, as using them may
+reduce interoperability.
```

Comment:
It seems like it's not undefined - calling it undefined seems to me to contradict the above "an embedded schema resource and its subschemas can be identified by JSON Pointer fragments relative to either its own canonical IRI, _or relative to any containing resource's IRI_".

This also leaves `[^8]` with no reference to it.

And, minor, but two spaces after a period isn't consistent with the rest of the document.

Reply:
It was discussed that we need to make pointer that cross boundaries an undefined behavior rather than allowing implementations to support it.  Of course, implementations can still support it if they want, but the spec should push schema authors a bit more forcefully toward using the proper `$id` instead.

I think the critical point here is "an embedded schema resource _and its subschemas_".  It's arguable that an embedded schema resource isn't strictly a subschema since it can be refactored out without any change in behavior.  If we need to clarify that, I'm happy to.

Reply:
Right, I get making this undefined is the main point of this PR (I edited my comment a bit to clarify that, had it slightly wrong initially ... I'm not sure this _should_ be undefined but if that is the way it is to be, okay).

To me "an embedded schema resource ... can be identified by JSON Pointer fragments ... relative to any containing resource's IRI" reads as defined behavior and not consistent with making it undefined some paragraphs later, and needs some change. I see your point on ambiguity whether "subschema" includes embedded schemas but I don't see language present that actually calls embedded schemas subschemas.

Reply:
I read that paragraph as informative, not prescriptive.  It's setting up the reasoning behind the requirement.

> Embedded schemas _can_ be referenced this way, but because reasons it's a bad idea to actually do it, so we're gonna say probably don't.  We're also letting implementations decide whether they want to support it, no pressure either way.